test/nodes/test_prompt_node.py

```diff
@@ -0,0 +1,355 @@
+        )
+
+    with pytest.raises(ValueError):
+        PromptTemplate(name="custom-task", prompt_text="Custom task: $param1", prompt_params=["param1", "param2"])
+
+
+def test_add_template_and_invoke(prompt_node):
+    tt = PromptTemplate(
+        name="sentiment-analysis-new",
+        prompt_text="Please give a sentiment for this context. Answer with positive, "
+        "negative or neutral. Context: $documents; Answer:",
+        prompt_params=["documents"],
+    )
+    prompt_node.add_prompt_template(tt)
+
+    r = prompt_node.prompt("sentiment-analysis-new", documents=["Berlin is an amazing city."])
+    assert r[0].casefold() == "positive"
+
+
+def test_on_the_fly_prompt(prompt_node):
+    tt = PromptTemplate(
+        name="sentiment-analysis-temp",
+        prompt_text="Please give a sentiment for this context. Answer with positive, "
+        "negative or neutral. Context: $documents; Answer:",
+        prompt_params=["documents"],
+    )
+    r = prompt_node.prompt(tt, documents=["Berlin is an amazing city."])
+    assert r[0].casefold() == "positive"
+
+
+def test_direct_prompting(prompt_node):
+    r = prompt_node("What is the capital of Germany?")
+    assert r[0].casefold() == "berlin"
+
+    r = prompt_node("What is the capital of Germany?", "What is the secret of universe?")
+    assert r[0].casefold() == "berlin"
+    assert len(r[1]) > 0
+
+    r = prompt_node("Capital of Germany is Berlin", task="question-generation")
+    assert len(r[0]) > 10 and "Germany" in r[0]
+
+    r = prompt_node(["Capital of Germany is Berlin", "Capital of France is Paris"], task="question-generation")
+    assert len(r) == 2
+
+
+def test_question_generation(prompt_node):
+    r = prompt_node.prompt("question-generation", documents=["Berlin is the capital of Germany."])
+    assert len(r) == 1 and len(r[0]) > 0
+
+
+def test_template_selection(prompt_node):
+    qa = prompt_node.set_default_prompt_template("question-answering")
+    r = qa(
+        ["Berlin is the capital of Germany.", "Paris is the capital of France."],
+        ["What is the capital of Germany?", "What is the capital of France"],
+    )
+    assert r[0].casefold() == "berlin" and r[1].casefold() == "paris"
+
+
+def test_has_supported_template_names(prompt_node):
+    assert len(prompt_node.get_prompt_template_names()) > 0
+
+
+def test_invalid_template_params(prompt_node):
+    with pytest.raises(ValueError):
+        prompt_node.prompt("question-answering", {"some_crazy_key": "Berlin is the capital of Germany."})
+
+
+def test_wrong_template_params(prompt_node):
+    with pytest.raises(ValueError):
+        # with don't have options param, multiple choice QA has
+        prompt_node.prompt("question-answering", options=["Berlin is the capital of Germany."])
+
+
+def test_run_invalid_template(prompt_node):
+    with pytest.raises(ValueError):
+        prompt_node.prompt("invalid-task", {})
+
+
+def test_invalid_prompting(prompt_node):
```

Comment:
not sure that we need test cases for this invalid input format. If you think we do, then I'd suggest we make the expected ValueError more specific.